source/UF/TruncatedTypes.lagda

```diff
@@ -0,0 +1,229 @@
+truncated-types-closed-under-equiv e =
+ truncated-types-are-closed-under-retracts (≃-gives-◁ e)
+
+\end{code}
+
+We can prove closure under embeddings as a consequence of the previous
+result.
+
+\begin{code}
+
+truncated-types-closed-under-embedding⁺ : {n : ℕ₋₂} {X : 𝓤 ̇ } {Y : 𝓥 ̇ }
+                                        → X ↪ Y
+                                        → Y is (n + 1) truncated
+                                        → X is (n + 1) truncated
+truncated-types-closed-under-embedding⁺ {𝓤} {𝓥} (e , is-emb) t x x' =
+ truncated-types-closed-under-equiv
+  (ap e , embedding-gives-embedding' e is-emb x x')
+  (t (e x) (e x'))
+
+truncated-types-closed-under-embedding : {n : ℕ₋₂}
+                                       → n ≥ −1
+                                       → {X : 𝓤 ̇ } {Y : 𝓥 ̇ }
+                                       → X ↪ Y
+                                       → Y is n truncated
+                                       → X is n truncated
+truncated-types-closed-under-embedding {𝓤} {𝓥} {succ n} _ =
+ truncated-types-closed-under-embedding⁺
+
+\end{code}
+
+Using closure under equivalence we can show closure under Σ and Π.
+
+\begin{code}
+
+truncated-types-closed-under-Σ : {n : ℕ₋₂} {X : 𝓤 ̇ } (Y : X → 𝓥 ̇ )
+                               → X is n truncated
+                               → ((x : X) → (Y x) is n truncated)
+                               → (Σ Y) is n truncated
+truncated-types-closed-under-Σ {𝓤} {𝓥} {−2} Y = Σ-is-singleton
+truncated-types-closed-under-Σ {𝓤} {𝓥} {succ n} Y l m (x , y) (x' , y') =
+ truncated-types-closed-under-equiv Σ-＝-≃
+  (truncated-types-closed-under-Σ
+   (λ p → transport Y p y ＝ y')
+   (l x x')
+   (λ p → m x' (transport Y p y) y'))
+
+truncated-types-closed-under-Π : {n : ℕ₋₂} {X : 𝓤 ̇ } (Y : X → 𝓥 ̇ )
+                               → ((x : X) → (Y x) is n truncated)
+                               → (Π Y) is n truncated
+truncated-types-closed-under-Π {𝓤} {𝓥} {−2} Y = Π-is-singleton fe
+truncated-types-closed-under-Π {𝓤} {𝓥} {succ n} Y m f g =
+ truncated-types-closed-under-equiv (happly-≃ fe)
+  (truncated-types-closed-under-Π (λ x → f x ＝ g x)
+  (λ x → m x (f x) (g x)))
+
+truncated-types-closed-under-→ : {n : ℕ₋₂} {X : 𝓤 ̇ } {Y : 𝓥 ̇ }
+                               → Y is n truncated
+                               → (X → Y) is n truncated
+truncated-types-closed-under-→ {𝓤} {𝓥} {n} {X} {Y} m =
+ truncated-types-closed-under-Π (λ - → Y) (λ - → m)
+
+\end{code}
+
+The subuniverse of types of n truncated types is defined as follows.
+
+\begin{code}
+
+𝕋 : ℕ₋₂ → (𝓤 : Universe) → 𝓤 ⁺ ̇
+𝕋 n 𝓤 = Σ X ꞉ 𝓤 ̇ , X is n truncated
+
+\end{code}
+
+From univalence we can show that 𝕋 n is n + 1 truncated,
+for all n : ℕ₋₂.
+
+\begin{code}
+
+truncation-levels-closed-under-≃⁺ : {n : ℕ₋₂} {X : 𝓤 ̇ } {Y : 𝓥 ̇ }
+                                  → Y is (n + 1) truncated
+                                  → (X ≃ Y) is (succ n) truncated
```

Comment:
Every time you write `succ n`, except in pattern matching, you should probably write `n + 1` like in line 92 above.

Oh, actually in line 203, you do write `n + 1`, but in line 204 you write `succ n`.

I suggest you make this modification everywhere in all files. There are several occurrences of this below.